### ISSUE TITLE
[Bug]: Proxy configuration does not work in version 0.0.18

### ISSUE THREAD

**Issue by manfwh** [web-infra-dev/rsbuild#435]
### 版本信息

```sh
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i7-4810MQ CPU @ 2.80GHz
    Memory: 1.73 GB / 15.67 GB
  Browsers:
    Edge: Chromium (119.0.2151.44)
    Internet Explorer: 11.0.19041.1
  npmPackages:
    @rsbuild/core: ^0.0.18 => 0.0.18
    @rsbuild/plugin-react: ^0.0.17 => 0.0.17
    @rsbuild/plugin-vue2: ^0.0.18 => 0.0.18
```

### 问题详情

![screenshot-20231110-113848](https://github.com/web-infra-dev/rsbuild/assets/23977810/c41fedef-e883-4854-97fa-eecd2e8feff8)
Proxy configuration does not work in version 0.0.18, other versions are normal
![screenshot-20231110-113703](https://github.com/web-infra-dev/rsbuild/assets/23977810/fdf7da4b-c7a1-4f31-82be-1283c007ae6e)

### 复现链接

https://github.com/manfwh/proxy-bug

### 复现步骤

1. node server.mjs
2. pnpm dev

**Comment by chenjiahan** [comment]
Fixed in https://github.com/web-infra-dev/rsbuild/pull/426

I will release a new version now.

**Comment by chenjiahan** [comment]
Released v0.0.19: https://github.com/web-infra-dev/rsbuild/releases/tag/v0.0.19

Please try it 😄
